tools/state-viewer/src/state_parts.rs

```diff
@@ -20,13 +21,21 @@ use std::path::{Path, PathBuf};
 use std::str::FromStr;
 use std::time::Instant;
 
+#[derive(clap::ArgEnum, Clone, Debug, Default)]
+pub(crate) enum ApplyAction {
+    #[default]
+    Apply,
+    Validate,
+    Print,
+}
+
 #[derive(clap::Subcommand, Debug, Clone)]
 pub(crate) enum StatePartsSubCommand {
     /// Apply all or a single state part of a shard.
     Apply {
```

Comment:
Looks good, the only concern I have here is readability. When I read "apply", I assume that something is applied/changed, but now we can have validate and print actions which doesn't modify anything.

I suggest to rename parent action to `Load`, similarly to python `json` load & dump API, and have `LoadAction` and `load_state_parts` to show that we read state parts from somewhere. Inside we have `Apply` action which means that we also apply part to state, etc.